rust/src/checkpoints.rs

```diff
@@ -388,12 +390,28 @@ fn parquet_bytes_from_state(state: &DeltaTableState) -> Result<bytes::Bytes, Che
     // Write the Checkpoint parquet file.
     let mut bytes = vec![];
     let mut writer = ArrowWriter::try_new(&mut bytes, arrow_schema.clone(), None)?;
+    let mut decoder = ReaderBuilder::new(arrow_schema)
+        .with_batch_size(CHECKPOINT_RECORD_BATCH_SIZE)
+        .build_decoder()?;
+
+    let mut buf = vec![];
+    for res in jsons {
+        let json = res?;
+        buf.write_all(serde_json::to_string(&json)?.as_bytes())?;
+    }
+    let mut consumed = 0;
```

Comment:
Same thing here, right? we shouldn't need to serialize to json string if in the end we are trying to make a parquet file?